packages/grid/x-data-grid-pro/src/components/GridColumnHeaders.tsx

```diff
@@ -49,61 +47,51 @@ const useUtilityClasses = (ownerState: OwnerState) => {
 };
 
 interface GridColumnHeadersPinnedColumnHeadersProps {
-  side: GridPinnedPosition;
+  side: GridPinnedColumnPosition;
   showCellVerticalBorder: boolean;
 }
 
-// Inspired by https://github.com/material-components/material-components-ios/blob/bca36107405594d5b7b16265a5b0ed698f85a5ee/components/Elevation/src/UIColor%2BMaterialElevation.m#L61
-const getOverlayAlpha = (elevation: number) => {
-  let alphaValue;
-  if (elevation < 1) {
-    alphaValue = 5.11916 * elevation ** 2;
-  } else {
-    alphaValue = 4.5 * Math.log(elevation + 1) + 2;
-  }
-  return alphaValue / 100;
-};
-
 const GridColumnHeadersPinnedColumnHeaders = styled('div', {
   name: 'MuiDataGrid',
   slot: 'PinnedColumnHeaders',
-  overridesResolver: (props, styles) => [
+  overridesResolver: (_, styles) => [
     { [`&.${gridClasses['pinnedColumnHeaders--left']}`]: styles['pinnedColumnHeaders--left'] },
     { [`&.${gridClasses['pinnedColumnHeaders--right']}`]: styles['pinnedColumnHeaders--right'] },
     styles.pinnedColumnHeaders,
   ],
 })<{ ownerState: OwnerState & GridColumnHeadersPinnedColumnHeadersProps }>(
   ({ theme, ownerState }) => ({
-    position: 'absolute',
+    position: 'sticky',
+    zIndex: 5,
     top: 0,
-    overflow: 'hidden',
-    zIndex: 1,
     display: 'flex',
     flexDirection: 'column',
+    boxSizing: 'border-box',
     boxShadow: theme.shadows[2],
-    backgroundColor: (theme.vars || theme).palette.background.default,
-    ...(theme.vars
-      ? {
-          backgroundImage: theme.vars.overlays?.[2],
-        }
-      : {
-          ...(theme.palette.mode === 'dark' && {
-            backgroundImage: `linear-gradient(${alpha('#fff', getOverlayAlpha(2))}, ${alpha(
-              '#fff',
-              getOverlayAlpha(2),
-            )})`,
-          }),
+    backgroundColor: 'var(--DataGrid-pinnedBackground)',
+    ...(ownerState.side === GridPinnedColumnPosition.LEFT && { left: 0 }),
+    ...(ownerState.side === GridPinnedColumnPosition.RIGHT && { right: 0 }),
+    [`&.${gridClasses['pinnedColumnHeaders--left']}`]: {
+      left: 0,
+      width: 'var(--DataGrid-leftPinnedWidth)',
```

Comment:
Is the width necessary here?

https://github.com/mui/mui-x/assets/13808724/51470117-6551-4d4d-92ee-e20659e479c4

Reply:
Removed.